pkg/cloud/retry_manager.go

```diff
@@ -0,0 +1,54 @@
+package cloud
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	"time"
+)
+
+const (
+	// retryMaxAttempt sets max number of EC2 API call attempts.
+	// Set high enough to ensure default sidecar timeout will cancel context long before we stop retrying.
+	retryMaxAttempt = 50
+	// retryMaxBackoffAdaptive sets max backoff delay between retry attempts.
+	// Set more aggressive than default of 20 seconds due to adaptive client-side throttling being enabled.
+	retryMaxBackoffAdaptive = 15 * time.Second
+)
+
+// newAdaptiveRetryer restricts attempts of API calls that recently hit throttle errors.
+func newAdaptiveRetryer() *retry.AdaptiveMode {
+	return retry.NewAdaptiveMode(func(ao *retry.AdaptiveModeOptions) {
+		ao.StandardOptions = append(ao.StandardOptions, func(so *retry.StandardOptions) {
+			so.MaxAttempts = retryMaxAttempt
+			so.MaxBackoff = retryMaxBackoffAdaptive
+		})
+	})
+}
+
+type retryManager struct {
+	createVolumeRetryer               *retry.AdaptiveMode
+	deleteVolumeRetryer               *retry.AdaptiveMode
+	attachVolumeRetryer               *retry.AdaptiveMode
+	detachVolumeRetryer               *retry.AdaptiveMode
+	modifyVolumeRetryer               *retry.AdaptiveMode
+	createSnapshotRetryer             *retry.AdaptiveMode
+	deleteSnapshotRetryer             *retry.AdaptiveMode
+	enableFastSnapshotRestoresRetryer *retry.AdaptiveMode
+
+	// Restrict unbatchable describe calls when rate-limited so that batched calls are more likely to succeed.
+	unbatchableDescribeVolumesModificationsRetryer *retry.AdaptiveMode
+}
+
+// newRetryManager initializes a new instance of retryManager that will dictate the retry strategies of EC2 API calls.
```

Comment:
Drop this comment, "newRetryManager initializes a new instance of retryManager" is obvious both from the code and the name of the function